src/query/query_executor.c

```diff
@@ -21843,11 +21843,24 @@ qexec_execute_build_indexes (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STA
 	}
     }
 
-  class_name = or_class_name (&class_record);
+  string = NULL;
+  alloced_string = 0;
+  error = or_class_name (&class_record, &string, &alloced_string);
+  if (error != NO_ERROR)
+    {
+      ASSERT_ERROR ();
+      GOTO_EXIT_ON_ERROR;
+    }
+  class_name = string;
+
   for (i = 0; i < rep->n_indexes; i++)
     {
       /* class name */
       db_make_string (out_values[0], class_name);
+      if (string != NULL && alloced_string == 1)
```

Comment:
why is the index of out_values is always 0 ?

Reply:
It follows the schema defined in the **pt_make_query_show_index** <sup>src/parser/parser_support.c</sup> function.
The table name must come with index 0.
```
const char *aliases[] = {
  "Table", "Non_unique", "Key_name", "Seq_in_index", "Column_name",
  "Collation", "Cardinality", "Sub_part", "Packed", "Null", "Index_type",
  "Func", "Comment", "Visible"
};
```

Reply:
OK, I see.